google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -160,7 +160,7 @@ private static String getValue(Label label, String resourceType) {
       case ContainerName:
         if (resourceType.equals("k8s_container")) {
           String hostName = System.getenv("HOSTNAME");
-          value = hostName.substring(0, hostName.indexOf("-"));
+          value = hostName.substring(0, hostName.lastIndexOf("-", hostName.lastIndexOf("-") - 1));
```

Comment:
Need unit testing for corner cases.

Reply:
This is not too easy to unit test, because everything is done via static methods and the hostname is read using System.getenv, which is brittle to fiddle with (probably that's the reason there are no really unit tests even for making sure the current code works.)

One thing I can do to test for edge cases would be to pull out the 

value = hostName.substring(0, hostName.lastIndexOf("-", hostName.lastIndexOf("-") - 1));

to a utility class (K8sContainerUtil?) and test only this one call via a unit test. Would that be acceptable? Or you have any better idea on how to go about testing this.

Reply:
Yes, I think it works. Thanks for your contribution, @wildhemp